sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -101,13 +104,16 @@ private[hive] object SparkSQLCLIDriver {
     SessionState.start(sessionState)
 
     // Clean up after we exit
-    Runtime.getRuntime.addShutdownHook(
-      new Thread() {
-        override def run() {
-          SparkSQLEnv.stop()
-        }
+    val cleanupHook = new Runnable {
```

Comment:
Does this need to be saved as a member? I think is can just be passed as an arg directly below.
Also `SparkSQLCLIDriver.SHUTDOWN_HOOK_PRIORITY` doesn't need to be qualified with class name.

@vanzin @piaozhexiu this is looking good and I wonder, are there other places that need a similar treatment? might as well zap all instances of this type of race.